classes/controller/ProductListingFrontController.php

```diff
@@ -300,6 +300,10 @@ protected function getProductSearchVariables()
 
         $query->setEncodedFacets($encodedFacets);
 
+        Hook::exec('actionProductSearchProviderRunQueryBefore', [
+            'query' => $query,
```

Comment:
```suggestion
            'query' => $query,
            'provider' => $provider,
            'context' => $context,
```

Reply:
Do you have confidence that this is needed? If you are not sure, then this is over engineering.